service/handlers.go

```diff
@@ -111,5 +111,28 @@ func (s *Service) infoHandler(w http.ResponseWriter, r *http.Request) {
 
 // healthHandler
 func (s *Service) healthHandler(w http.ResponseWriter, r *http.Request) {
-	w.Write([]byte("OK"))
+	s.StateProbes.MuHealthy.Lock()
+	defer s.StateProbes.MuHealthy.Unlock()
+
+	if s.StateProbes.IsHealthy {
```

Comment:
I'd prefer to have fail-fast strategy. First handle errors and successful resolution should be at the end of function